example/extending/newlang/5-hello-python/build.mill

```diff
@@ -0,0 +1,158 @@
+// does not come bundled with https://www.python.org[Python] integration, so here we begin setting
+// one up from first principles using the Basic `Python3`(Already Present in OS) command Line.
+
+// === Python Setup
+//
+// First, we need to use the `Python3` CLI tool to create virtual environment and
+// install `mypy` for Type Checking using the official Python Virtual Environment
+// https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments[Documentation].
+
+// `MyPy` is a static type checker that uses Python `type hints` to enforce `type correctness` and `catch errors` during development.
+
+package build
+import mill._
+
+def createVenv: T[PathRef] = Task {
+  val venvDir = T.dest / "venv"
+
+  os.call(
+    Seq(
+      "python3",
+      "-m",
+      "venv",
+      venvDir.toString
+    )
+  )
+
+  PathRef(venvDir)
+}
+
+def setup: T[PathRef] = Task {
+  // creating virtual environment
+  val pythonVenv = createVenv().path / "bin" / "python3"
+  
+  // installing mypy for Type Checking
+  os.call(
+    Seq(
+      pythonVenv.toString,
+      "-m",
+      "pip",
+      "install",
+      "mypy"
+    )
+  )
+  PathRef(pythonVenv)
+}
+
+// The `createVenv` task ensures the Python virtual environment creation process using the official documentaion.
+// After this, we need to install python `mypy` library for Type Checking and Error Checking before actual run of the code, 
+// using the offical https://mypy.readthedocs.io/en/stable[Documentation] for `mypy` 
+
+// === Defining our Sources
+
+// Next, We define the `sources` task to help gather source files for processing. 
+// `sources` sets a base directory for Python source files and enable easy access to Python files in a project,
+// making it straightforward to retrieve all relevant code files for tasks like building, testing, or analysis.
+
+def sources: T[PathRef] = Task.Source(millSourcePath / "src")
+
+// === Type Checking
+
+// The `mainFileName` Task provides the name for the main python script file for running and 
+// The `typeCheck` Task  ensures that the python setup is set and Checks the main file for error free code.
+// This task is considered as a part of code which is responsible for Error Checking of the Main Script 
+// and setting up setup for python script runable environment.
+
+def mainFileName: T[String] = Task { "main.py" }
+def typeCheck: T[PathRef] = Task {
+  val pythonVenv = setup().path
+
+  os.call(
+    Seq(
+      pythonVenv.toString,
+      "-m",
+      "mypy",
+      "--strict",
+      s"${sources().path / mainFileName()}"
+    ),
+    stdout = os.Inherit
+  )
+
+  PathRef(pythonVenv)
+}
+
+// At this point, we have a minimal working build, with a build graph that looks like this:
+//
+// ```graphviz
+// digraph G {
+//   rankdir=LR
+//   node [shape=box width=0 height=0 style=filled fillcolor=white]
+//   createVenv -> setup -> typeCheck
+//   sources -> typeCheck
+//   mainFileName -> typeCheck
+// }
+// ```
```

Comment:
We can probably skip this diagram since it's not yet interesting; the one below is enough